docs/Customization/Custom-Core.rst

```diff
@@ -0,0 +1,267 @@
+      val nLocalInterrupts: Int           // # of local interrupts (see SiFive interrupt cookbook)
+      val nPMPs: Int                      // # of Physical Memory Protection units
+      val pmpGranularity: Int             // Size of the smallest unit of region for PMP unit (must be power of 2)
+      val nBreakpoints: Int               // # of hardware breakpoints supported (in RISC-V debug specs)
+      val useBPWatch: Boolean             // Support hardware breakpoints
+      val nPerfCounters: Int              // # of supported performance counters
+      val haveBasicCounters: Boolean      // Support basic counters defined in the RISC-V counter extension 
+      val haveFSDirty: Boolean            // If true, the core will set FS field in mstatus CSR to dirty when appropriate
+      val misaWritable: Boolean           // Support writable misa CSR (like variable instruction bits)
+      val haveCFlush: Boolean             // Rocket specific: enables Rocket's custom instruction extension to flush the cache
+      val nL2TLBEntries: Int              // # of L2 TLB entries
+      val mtvecInit: Option[BigInt]       // mtvec CSR (of V extension) initial value
+      val mtvecWritable: Boolean          // If mtvec CSR is writable
+
+      // Normally, you don't need to change these values (except lrscCycles) 
+      def customCSRs(implicit p: Parameters): CustomCSRs = new CustomCSRs
+
+      def hasSupervisorMode: Boolean = useSupervisor || useVM
+      def instBytes: Int = instBits / 8
+      def fetchBytes: Int = fetchWidth * instBytes
+      // Rocket specific: Longest possible latency of Rocket core D1 cache. Simply set it to the default value 80 if you don't use it.
+      def lrscCycles: Int
+
+      def dcacheReqTagBits: Int = 6
+
+      def minFLen: Int = 32
+      def vLen: Int = 0
+      def sLen: Int = 0
+      def eLen(xLen: Int, fLen: Int): Int = xLen max fLen
+      def vMemDataBits: Int = 0
+    }
+
+    case class FPUParams(
+      minFLen: Int = 32,          // Minimum floating point length (no need to change) 
+      fLen: Int = 64,             // Maximum floating point length, use 32 if only single precision is supported
+      divSqrt: Boolean = true,    // Div/Sqrt operation supported
+      sfmaLatency: Int = 3,       // Rocket specific: Fused multiply-add pipeline latency (single precision)
+      dfmaLatency: Int = 4        // Rocket specific: Fused multiply-add pipeline latency (double precision)
+    )
+
+Most of the fields here are originally designed for the Rocket core and thus contain some implementation-specific details, but 
+many of them are general enough to be useful for other cores. It is strongly recommended to use these fields instead
+of creating your own custom fields when applicable. 
+
+You will also need a ``CanAttachTile`` class to add the tile config into the config system, with the following format:
```

Comment:
What is "CanAttachTile" going to be used for?

Reply:
Chipyard will instantiate a tile for every ``CanAttachTile`` found under ``TilesLocated(InSubsystem)``.

Reply:
Explanation added.